libs/ts-rest/core/src/lib/client.spec.ts

```diff
@@ -100,29 +105,40 @@ const postsRouter = c.router({
 });
 
 // Three endpoints, two for posts, and one for health
-export const router = c.router({
-  posts: postsRouter,
-  health: {
-    method: 'GET',
-    path: '/health',
-    responses: {
-      200: c.response<{ message: string }>(),
+export const router = c.router(
+  {
+    posts: postsRouter,
+    health: {
+      method: 'GET',
+      path: '/health',
+      responses: {
+        200: c.response<{ message: string }>(),
+      },
     },
-  },
-  upload: {
-    method: 'POST',
-    path: '/upload',
-    body: c.body<{ file: File }>(),
-    responses: {
-      200: c.response<{ message: string }>(),
+    upload: {
+      method: 'POST',
+      path: '/upload',
+      body: c.body<{ file: File }>(),
+      responses: {
+        200: c.response<{ message: string }>(),
+      },
+      contentType: 'multipart/form-data',
     },
-    contentType: 'multipart/form-data',
   },
-});
+  {
+    baseHeaders: z.object({
```

Comment:
So in this case, these baseHeaders will only apply to `upload` and `health`, but not `posts` correct?